experiments/customui/parent.js

```diff
@@ -250,21 +250,46 @@ var ex_customui = class extends ExtensionCommon.ExtensionAPI {
       return result;
     };
 
+    // Enables dynamic width and fixed heigth for a WebExtension frame
+    const setWebextFrameSizesForSidebar = function(frame, options) {
+      setWebextFrameSizesForBox(frame, options, true, "244px", false, "100%")
+    }
+
     // Enables dynamic height and fixed 100% width for a WebExtension frame
     const setWebextFrameSizesForVerticalBox = function(frame, options) {
-      frame.width = "100%";
-      frame.height = (options.height || 100) + "px";
-      frame.style.display = options.hidden ? "none" : "block";
+      setWebextFrameSizesForBox(frame, options, false, "100%", true, "100px");
+    }
+
+    const setWebextFrameDynamicDimension = function(frame, options, dimensionName, defaultValue) {
+      frame[dimensionName] = (options[dimensionName] || defaultValue) + "px";
       frame.addCustomUILocalOptionsListener(lOptions => {
-        if (typeof lOptions.height === "number") {
-          frame.height = lOptions.height + "px";
-          frame.style.height = frame.height;
+        if (typeof lOptions[dimensionName] === "number") {
+          frame[dimensionName] = lOptions[dimensionName] + "px";
+          frame.style[dimensionName] = frame[dimensionName];
         }
         if (typeof lOptions.hidden === "boolean") {
           frame.style.display = lOptions.hidden ? "none" : "block";
         }
       });
     };
+
+    const setWebextFrameFixedDimension = function(frame, dimensionName, value) {
```

Comment:
Spreading things out over tiny functions seems less intuitive due to minor differences in expectation: `value` here is a CSS string, while `defaultValue` above is a pixel number. Adding a comment before each function helps to mitigate such cases to some extent (and makes them more visible to you while programming).

Reply:
done, the main goal I had in mind was reuse. So there was no code copied for sidebar and verticalbox.
The arguments were indeed not clear. I added comments